Automattic/newspack-theme
# Secondary Menu: Add support for dropdown menus

### All Submissions:


### Changes proposed in this Pull Request:

This PR makes it possible to add multi-level menus to the Secondary Menu location. It also refactors the original dropdown styles, tying them to its own class (`dd-menu`) so it will be easier to add to other locations, if needed.

Closes #721.

### How to test the changes in this Pull Request:

1. Set your test site to use default colours.
2. Add a menu with multiple levels to both the primary and secondary menu locations.
3. Note how the primary menu dropdowns look:

<img width="1252" alt="image" src="https://user-images.githubusercontent.com/177561/90208833-a457a500-dd9e-11ea-9b41-504711f9a91c.png">

4. Apply the PR and run `npm run build`.
5. Confirm that the secondary menu dropdown is appearing, and looks like the primary menu dropdown (which should be unchanged):

<img width="866" alt="image" src="https://user-images.githubusercontent.com/177561/90208891-ce10cc00-dd9e-11ea-9e1b-5403ecc7437b.png">

6. Check on mobile and confirm both menus are displaying multiple levels, with no issues:

<img width="378" alt="image" src="https://user-images.githubusercontent.com/177561/90208919-e385f600-dd9e-11ea-97af-6324d6ac9809.png">

7. Change your custom colours to a light colour; confirm that the menus on desktop and mobile are picking up the colours, and have sufficient contrast.
8. Change your custom colours to a dark colour; confirm that the menus on desktop and mobile are picking up the colours, and have sufficient contrast.

### Other information:
